code/game/machinery/sprinkler.dm

```diff
@@ -0,0 +1,214 @@
+	density = FALSE
+	deployable = /obj/item/deployable/sprinkler
+	volume = 10
+	var/dirty = FALSE
+	var/cooldown = 0
+
+/obj/machinery/power/plumbing/sprinkler/Initialize()
+	create_reagents(volume, AMOUNT_VISIBLE)
+	update_icon()
+	return ..()
+
+/obj/machinery/power/plumbing/sprinkler/ComponentInitialize()
+	AddComponent(/datum/component/plumbing/output)
+
+/obj/machinery/power/plumbing/sprinkler/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>It is [anchored? "anchored" : "not anchored"].</span>"
+	. += "<span class='notice'>[dirty? "It's dirty" : "It's clean"].</span>"
+
+/obj/machinery/power/plumbing/sprinkler/default_unfasten_wrench(mob/user, obj/item/I, time = 5)
+	..()
+	if(anchored)
+		START_PROCESSING(SSobj, src)
+	else
+		STOP_PROCESSING(SSobj, src)
+
+/obj/machinery/power/plumbing/sprinkler/process()
+	///find sparks
+	var/turf/T = get_turf(loc)
+	var/obj/effect/particle_effect/sparks/S = locate() in T.contents
+	if(S)
+		dispense()
+	if(prob(10))
+		///find dirt
+		var/obj/effect/decal/cleanable/D = locate() in T.contents
+		if(D)
+			dirty = TRUE
+			STOP_PROCESSING(SSobj, src)
+			update_icon()
+
+/obj/machinery/power/plumbing/sprinkler/update_icon()
+	. = ..()
+	cut_overlays()
+	if(dirty)
+		add_overlay(mutable_appearance(icon,"sprinkler_dirty"))
+	if(!anchored)
+		add_overlay(mutable_appearance(icon,"sprinkler_connection"))
+
+/obj/machinery/power/plumbing/sprinkler/plunger_act(obj/item/plunger/P, mob/living/user)
+	. = ..()
+	if(!dirty)
+		to_chat(user, "<span class='notice'>\The [src] is clean, there is no need to plunger it.</span>")
+		return
+	if(!working)
+		to_chat(user, "<span class='notice'>You start plunging  \the [src]...</span>")
+		working = TRUE
+		if(do_after(user, 50, target = src))
+			to_chat(user, "<span class='notice'>You have plunged \the [src].</span>")
+			dirty = FALSE
+			START_PROCESSING(SSobj, src)
+			update_icon()
+		else
+			working = FALSE
+
+/obj/machinery/power/plumbing/sprinkler/interact(mob/user)
+	if(!allowed(user))
+		to_chat(user, "<span class='warning'>Access denied.</span>")
+		return
+	dispense()
+	log_game("[user] has manually actived the sprinkler in [AREACOORD(src)]")
+
+/obj/machinery/power/plumbing/sprinkler/fire_act()
+	. = ..()
+	dispense()
+
+/obj/machinery/power/plumbing/sprinkler/proc/dispense()
+	if(cooldown < world.time && !dirty && anchored)
+		playsound(src, 'sound/machines/beep.ogg', 100, 1)
+		var/datum/effect_system/smoke_spread/chem/smoke_machine/smoke = new()
+		smoke.set_up(reagents, 5, 8,  get_turf(src))
+		smoke.start()
+		cooldown = world.time + 200
+		log_game("A sprinkler has been activated in [AREACOORD(src)]")
+
+/obj/machinery/power/plumbing/sprinkler/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	..()
+
+/obj/item/deployable
+	name = "deployable thing"
+	desc = "A self-deploying thing, it shouldn't be here."
+	icon = 'icons/obj/plumbing/plumbers.dmi'
+	icon_state = "sprinkler_d"
+	var/deploying = FALSE
+	var/obj/machinery/power/plumbing/result
```

Comment:
this looks yikes. the class implies it can deploy anything but its plumbing only.